core/src/dird/authenticate_console.cc

```diff
@@ -407,10 +406,12 @@ static ConsoleAuthenticator* CreateConsoleAuthenticator(UaContext* ua)
 
 bool AuthenticateConsole(UaContext* ua)
 {
-  if (NumberOfConsoleConnectionsExceeded()) {
+  uint32_t ConsoleConnections = CurrentNumberOfConsoleConnections();
+  if (ConsoleConnections >= me->MaxConsoleConnections) {
     Emsg0(M_ERROR, 0,
           _("Number of console connections exceeded "
-            "MaximumConsoleConnections\n"));
+            "Maximum :%u, Current: %u\n"),
```

Comment:
Is this run on a single thread ? Otherwise this does not really ensure that the invariant
`ConsoleConnections >= me->MaxConsoleConnectinos` hold.  But it should be true in most circumstances (think of the scenario where two consoles want to connect "at the same time" while there is only one spot left.